### PR TITLE
[CSFS-153] Article Publishing Endpoints

### PR DESCRIPTION
**Pull request by kojusola**
## Task: `Created Endpoints for Article Publishing`
- `In addition to creating endpoints, I created documentation for the endpoints, the link can be found in the ApiDocumentation.txt file`

### CODE REVIEW

File: submissions/adeolaFafemi/phpSlim/app/constants/constants.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+
+namespace App\Constants;
+class constants
+{
+    const CONTENT_TYPE_HEADER = 'Content-Type';
+    const APPLICATION_JSON = 'Application/json';
+}
```
Comment (adaugochi) [inline submissions/adeolaFafemi/phpSlim/app/constants/constants.php:9]: Remove spaces below

File: submissions/adeolaFafemi/phpSlim/app/controllers/ArticlesController.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        $response->getBody()->write(json_encode([ 'status'=> 'success','data' => $article]));
+        return $response -> withHeader(Constants::CONTENT_TYPE_HEADER, Constants::APPLICATION_JSON)->withStatus(200);
+    }
+    public function deleteArticle (Request $request, Response $response, $args)
+    {
+        $id = $args['id'];
+        $article = Article::find($id);
+        if(!$article){
+            $response->getBody()->write(json_encode(["error"=> "Article not found"]));
+            return $response -> withHeader('Content-Type', 'application/json')->withStatus(404);
```
Comment (adaugochi) [inline submissions/adeolaFafemi/phpSlim/app/controllers/ArticlesController.php:49]: Use the constant here

File: submissions/adeolaFafemi/phpSlim/app/controllers/ArticlesController.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        return $response -> withHeader(Constants::CONTENT_TYPE_HEADER, Constants::APPLICATION_JSON)->withStatus(200);
+    }
+    public function publishArticle (Request $request, Response $response, $args)
+    {
+        // $data = $request ->getParsedBody();
+        $id = $args['id'];
+        $article = Article::find($id);
+        if(!$article){
+            $response->getBody()->write(json_encode(["error"=> "Article not found"]));
+            return $response -> withHeader('Content-Type', 'application/json')->withStatus(404);
```
Comment (adaugochi) [inline submissions/adeolaFafemi/phpSlim/app/controllers/ArticlesController.php:62]: same here

File: submissions/adeolaFafemi/phpSlim/app/controllers/ArticlesController.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        $response->getBody()->write(json_encode([ 'status'=> 'success','data' => $articles]));
+        return $response -> withHeader(Constants::CONTENT_TYPE_HEADER, Constants::APPLICATION_JSON)->withStatus(200);
+    }
+    public function singleArticle (Request $request, Response $response, $args)
+    {
+        $id = $args['id'];
+        $article = Article::find($id);
+        if(!$article){
+            $response->getBody()->write(json_encode(["error"=> "User not found"]));
+            return $response -> withHeader('Content-Type', 'application/json')->withStatus(404);
```
Comment (adaugochi) [inline submissions/adeolaFafemi/phpSlim/app/controllers/ArticlesController.php:85]: same here

**Review by adaugochi — CHANGES_REQUESTED** [review]
I checked your API documentation, I didn't see your  Example response. it means you didn't save your response you got. Kindly follow the article and give a thumbs up if you find it helpful

File: submissions/adeolaFafemi/phpSlim/app/controllers/ArticlesController.php (hunk 4 above)
Comment (kojusola) [reply submissions/adeolaFafemi/phpSlim/app/controllers/ArticlesController.php]: Corrected.
Thank you.

File: submissions/adeolaFafemi/phpSlim/app/constants/constants.php (hunk 1 above)
Comment (adaugochi) [reply submissions/adeolaFafemi/phpSlim/app/constants/constants.php:9]: Remove spaces below

**Review by adaugochi — CHANGES_REQUESTED** [review]
Remove `composer.lock` file, it shouldn't be part of your PR

**Review by adaugochi — CHANGES_REQUESTED** [review]
Remove composer.lock file
image

**Comment by kojusola** [comment]
It should be out now. 
Thank you.

**Review by adaugochi — APPROVED** [review]

**Pull request merged**